invocation.md

```diff
@@ -0,0 +1,1170 @@
+          "do": "crud/create",
+          "input": {
+            "headers": {
+              "content-type": "application/json"
+            },
+            "payload": {
+              "title": "How UCAN Tasks Changed My Life",
+              "body": "This is the story of how one spec changed everything...",
+              "topics": ["authz", "journal"],
+              "draft": true
+            }
+          },
+          "s": {
+            "/": {
+              "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+            }
+          },
+          "prf": [
+            {
+              "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a"
+            }
+          ]
+        },
+        "ok"
+      ]
+    },
+    "sendEmail": {
+      "<-": [
+        {
+          "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+          "aud": "did:web:ucan.run",
+          "with": "mailto:akiko@example.com",
+          "do": "msg/send",
+          "input": {
+            "to": ["boris@example.com", "carol@example.com"],
+            "subject": "Coffee",
+            "body": "Hey you two, I'd love to get coffee sometime and talk about UCAN Tasks!"
+          },
+          "s": {
+            "/": {
+              "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+            }
+          },
+          "prf": [
+            {
+              "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a"
+            }
+          ]
+        },
+        "ok"
+      ]
+    },
+    "sendTextMessage": {
+      "<-": [
+        {
+          "/": "bafybeibwlfwol5bdwj75hdqs3liv6z7dyqwtd2t4ovvgncv4ixkaxlkfle"
+        },
+        "ok"
+      ]
+    }
+  },
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  }
+}
+```
+
+# 4 Receipt
+
+An Invocation Receipt is an attestation of the Result of an Invocation. A Receipt MUST be signed by the Executor (the `aud` of the associated UCANs) or it's delegate, in which case proof of delegation (of the invoked capability) from Executor to the Signer (the `iss` of the receipt) MUST be provided in `prf`.
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts MUST use the same version as the invocation that they contain.
+
+## 4.1 Schema
+
+```ipldsch
+type Receipt<In, Out> struct {
+  job     &Invocation<In>
+
+  # output of the invocation
+  out     Out
+
+  # Related receipts
+  origin optional &Receipt<In, Any>
+
+  # All the other metadata
+  meta    { String: Any }
+
+  # Principal that issued this receipt
+  iss     Principal
+
+  # Proof that `iss` was authorized to by invocation `aud` to issue
+  # receipts. Can be omitted if `job.aud === this.iss`
+  prf     [&UCAN] implicit ([])
```

Comment:
Actually same for this, though the spec may not be entirely up to date: won't you be able to find this in the Invocation? Something like `receipt.job.scope.prf`?

Reply:
So both `iss` and `prf` are to support the way our system is designed. Specifically client will sent invocation to `did:web:web3.storage` but receipt will be issued by `did:key:zKeyInJanuary` and `prf` will include delegation chain `did:web:web3.storage` to `did:key:zKeyInJanuary`.

Also note that delegation will be pre-arranged as opposed to created on invocation, which is why we'd like to capture proof chain that actor that issued receipt was acting on behalf of executor.

`prf` is already optional and I don't mind making `iss` optional either in which case it would imply to be the `aud` of the invocation.

Reply:
> won't you be able to find this in the Invocation? Something like `receipt.job.scope.prf`?

You won't be able to find it in the invocation because client submitting invocation will not know what keys are in rotation or if we have a pool of workers that pick up task which one of would.